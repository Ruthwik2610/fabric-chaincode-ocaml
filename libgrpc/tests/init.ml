open GRPC

module Core = struct
  open Core

  let () =
    let slice = slice_of_string "Test" in
    print_endline @@ string_of_slice slice

  let () = print_endline (grpc_version_string ())

  let () = print_endline (grpc_g_stands_for ())
end

let server = create_server ~listening:"localhost:30300" ()

let () =
  match wait_call ~timeout:2L server with
  | TIMEOUT -> print_endline "wait_call timeout"
  | CALL _ -> print_endline "wait_call succeeded"

let () = destroy_server server
