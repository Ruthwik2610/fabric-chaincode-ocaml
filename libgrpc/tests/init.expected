Test
10.0.0
gradius
wait_call timeout
